src/setup_payload/tests/run_python_setup_payload_gen_test.py

```diff
@@ -0,0 +1,129 @@
+#!/usr/bin/env python3
+
+# Copyright (c) 2023 Project CHIP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import random
+import re
+import subprocess
+import sys
+
+CHIP_TOPDIR = os.path.dirname(os.path.realpath(__file__))[:-len(os.path.join('src', 'setup_payload', 'tests'))]
+sys.path.insert(0, os.path.join(CHIP_TOPDIR, 'src', 'setup_payload', 'python'))
+from generate_setup_payload import INVALID_PASSCODES, CommissioningFlow, SetupPayload  # noqa: E402
+
+
+def payload_param_dict():
+    return {
+        'Version': None,
+        'VendorID': None,
+        'ProductID': None,
+        'Custom flow': None,
+        'Discovery Bitmask': None,
+        'Short discriminator': None,
+        'Long discriminator': None,
+        'Passcode': None
+    }
+
+
+def remove_escape_sequence(data):
+    ansi_escape = re.compile(r'\x1B(?:[@-Z\\-_]|\[[0-?]*[ -/]*[@-~])')
+    result = ansi_escape.sub('', data)
+    return result
+
+
+def parse_setup_payload(chip_tool, payload):
+    cmd_args = [chip_tool, 'payload', 'parse-setup-payload', payload]
+    data = subprocess.check_output(cmd_args).decode('utf-8')
+    data = remove_escape_sequence(data)
+    parsed_params = payload_param_dict()
+    for key in parsed_params:
+        k_st = data.find(key)
+        if k_st == -1:
+            continue
+
+        # 1 is for ":"
+        k_end = k_st + len(key) + 1
+
+        k_nl = data.find('\n', k_end)
+        parsed_params[key] = data[k_end:k_nl].split()[0]
+
+    return parsed_params
+
+
+def generate_payload_params():
+    payload_params = payload_param_dict()
+    payload_params['Version'] = 0
+    payload_params['VendorID'] = random.randint(0, 0xFFFF)
```

Comment:
The randomness here will make it hard to tell if this ever fails; people will just retrigger and it will pass...

What would be better is having a set of possible "representative" values (2-4) for the things that are being randomly generated here and trying either all combinations or if that's too many options and takes too long to run some subset defined somehow.

Reply:
Now I have predefined some data set and checking against them.